omnibus/config/templates/updater/updater-admin-exec.sh.erb

```diff
@@ -0,0 +1,136 @@
+        echo "failed datadog start check"
+        return 1
+    fi
+    if [ "${#unit}" -gt 50 ]; then
+        error_message="unit name longer then 50"
+        echo "greater then"
+        return 1
+    fi
+    tmp_str="${unit}"
+    while [ -n "$tmp_str" ]; do
+        rest="${tmp_str#?}"
+        first_character="${tmp_str%"$rest"}"
+        tmp_str="$rest"
+        case "$first_character" in
+            [abcdefghijklmnopqrstuvwxyz-])
+                ;;
+            ".")
+                ;;
+            *)
+                error_message="invalid unit name: $unit"
+                echo "failed"
+                return 1
+                ;;
+        esac
+    done
+    echo "success"
+    return 0
+}
+
+handle_command() {
+    command="$1"
+    case "$command" in
+        "stop" | "enable" | "disable")
+            if [ "$#" -ne 2 ]; then
+                error_message="missing arguments"
+                return 1
+            fi
+            unit="$2"
+            assert_unit "$unit" || return 1
+            if ! systemctl "$command" "$unit"; then
+                error_message="failed to $command unit: $unit"
+                return 1
+            fi
+            return 0
+            ;;
+        "start")
+            if [ "$#" -ne 2 ]; then
+                error_message="missing arguments"
+                return 1
+            fi
+            unit="$2"
+            assert_unit "$unit" || return 1
+            # --no-block is used to avoid blocking on oneshot executions
+            if ! systemctl "$command" "$unit" --no-block ; then
+                error_message="failed to $command unit: $unit"
+                return 1
+            fi
+            return 0
+            ;;
+        "reload")
+            if ! "systemctl daemon-reload"; then
+                error_message="failed to reload: $unit"
+                return 1
+            fi
+            return 0
+            ;;
+        "load-unit")
+            if [ "$#" -ne 2 ]; then
+                error_message="missing arguments"
+                return 1
+            fi
+            unit="$2"
+            assert_unit "$unit" || return 1
+            echo "added"
+            cp "$unit_path/$unit" "$systemd_path/$unit"
+            echo "it passes"
+            return 0
+            ;;
+        "remove-unit")
+            if [ "$#" -ne 2 ]; then
+                error_message="missing arguments"
+                return 1
+            fi
+            unit="$2"
+            assert_unit "$unit" || return 1
+            rm "$systemd_path/$unit"
+            echo "removed"
+            return 0
+            ;;
+        *)
+            error_message="error: Invalid command: $command"
+            return 1
+            ;;
+    esac
+}
+
+while true; do
+    echo "waiting"
+    if ! read -r command < "$in_fifo"; then
+        rm "$out_fifo" || true
+        exit 0
+    fi
+    echo "read $command"
+    # shellcheck disable=SC2086
+    handle_command $command
```

Comment:
todo enforce normalisation of command to avoid word splitting
($command is expanded to pass N parameters to handle_command()